### PR TITLE
fix: added custom category filter to allow sorting by name

### PR DESCRIPTION
**Pull request by Bernardvdv**

### CODE REVIEW

File: djangocms_alias/admin.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
     alias_content_admin_classes.insert(0, ExtendedVersionAdminMixin)
     alias_content_admin_list_display = ('name', 'get_category',)
     alias_content_admin_list_filter = (SiteFilter, LanguageFilter, UnpublishedFilter)
```
Comment (Aiky30) [inline djangocms_alias/admin.py:40]: `alias_content_admin_list_filter` appears to be duplicated historically, we should clean that.

Comment (Aiky30) [inline djangocms_alias/admin.py:40]:
```suggestion
```

File: djangocms_alias/filters.py [hunk 2]
```diff
@@ -82,6 +85,37 @@ def choices(self, changelist):
...
+class CategoryFilter(admin.SimpleListFilter):
+    title = _("Category")
+    parameter_name = CATEGORY_FILTER_PARAM
+
+    def lookups(self, request, model_admin):
+        qs = model_admin.get_queryset(request)
+        cat_id = qs.values_list('alias__category', flat=True).distinct()
+        cat = Category.objects.filter(pk__in=cat_id).order_by('translations__name')
+        for obj in cat:
+            yield str(obj.pk), smart_text(obj)
```
Comment (Aiky30) [inline djangocms_alias/filters.py:97]: smart_text is deprecated in django 3: https://docs.djangoproject.com/en/4.0/releases/3.0/#features-deprecated-in-3-0

some help to replace: https://docs.djangoproject.com/en/3.2/_modules/django/utils/encoding/#smart_text

File: djangocms_alias/filters.py [hunk 3]
```diff
@@ -82,6 +85,37 @@ def choices(self, changelist):
...
 
 
+class CategoryFilter(admin.SimpleListFilter):
+    title = _("Category")
+    parameter_name = CATEGORY_FILTER_PARAM
+
+    def lookups(self, request, model_admin):
+        qs = model_admin.get_queryset(request)
+        cat_id = qs.values_list('alias__category', flat=True).distinct()
+        cat = Category.objects.filter(pk__in=cat_id).order_by('translations__name')
```
Comment (Aiky30) [inline djangocms_alias/filters.py:95]:
```suggestion
        # Ensure the category is ordered by the name alphabetically by default
        cat = Category.objects.filter(pk__in=cat_id).order_by('translations__name')
```

Try to add comments to reinforce why something exists.

**Review by Aiky30 — CHANGES_REQUESTED** [review]
Minor changes, smart_text being the biggest

**Review by adam-murray — APPROVED** [review]

File: tests/test_admin_filters.py [hunk 4]
```diff
@@ -332,3 +334,76 @@ def test_category_filter_with_verisoning(self):
...
+
+        version_admin = admin.site._registry[AliasContent]
+        category_filter = CategoryFilter(None, {"category": ""}, AliasContent, version_admin)
+
+        # Get the first choice in the filter lookup object
+        first_lookup_value = category_filter.lookup_choices[0][1]
+        # Lookup choices should only display the category linked to content
+        self.assertEqual(len(category_filter.lookup_choices), 1)
+        # Lookup value should match the category name linked to alias content
+        self.assertEquals(
```
Comment (Aiky30) [inline tests/test_admin_filters.py:403]: assertEquals is deprecated: https://docs.python.org/3/library/unittest.html#deprecated-aliases
```suggestion
        self.assertEqual(
```

File: tests/test_admin_filters.py [hunk 5]
```diff
@@ -332,3 +334,76 @@ def test_category_filter_with_verisoning(self):
...
+            name="EN Alias Content two",
+            language="en",
+        )
+
+        version_admin = admin.site._registry[AliasContent]
+        category_filter = CategoryFilter(None, {"category": ""}, AliasContent, version_admin)
+        # Get the first choice in the filter lookup object
+        first_lookup_value = category_filter.lookup_choices[0][1]
+        # Lookup value should match the category name linked to alias content
+        self.assertEquals(
```
Comment (Aiky30) [inline tests/test_admin_filters.py:368]:
```suggestion
        self.assertEqual(
```

assertEquals is deprecated: https://docs.python.org/3/library/unittest.html#deprecated-aliases

**Review by Aiky30 — CHANGES_REQUESTED** [review]
2 minor changes for deprecation. You can run the depecation and warnings checks in tests by add ing -Wd to the test parameters.

**Review by Aiky30 — APPROVED** [review]

**Pull request merged**